File: day1/main.js
const fs = require('fs');
const readline = require('readline')

const createElf = () => {
    let rl = readline.createInterface({
        input: fs.createReadStream("./input"),
        output: process.stdout,
        terminal: false
    });
    let elves = []
    let current = 0;
    let max = 0;
    rl.on('line', text => {
        if (text !== "\n") {
            current = current + parseInt(text);
            console.log(current);
        } else {
            max = Math.max(current, max);
        }
    })
}
createElf();

File: day1/main.py
from heapq import nlargest

def main():
    f = open("input", "r")
    lines = f.readlines()
    elves = []
    current = 0
    max1 = 0
    for line in lines:
        if line.find("\n") != 0:
            current += int(line)
        else:
            elves.append(current)
            max1 = max(current, max1)
            current = 0
    

    top3sum = sum(nlargest(3, elves))
    print(top3sum)
    print(max1)

main()